### DISCUSSION TITLE
[pnpm v7] cant find @types when it is installed by a dependency, for example @testing-library/jest-dom

### DISCUSSION THREAD

**Discussion by leopen-hu** [pnpm/pnpm#4723] [Q&A]
I updated my pnpm to v7 and missed some types for Typescript.

You can create a `create-react-app` typescript app and then use pnpm as packages manager to confirm that:
```
npx create-react-app pnpm-demo-v7 --template typescript
cd pnpm-demo-v7

# runs well
npm run build

rm-rf node_modules
pnpm i

# Error: Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'
pnpm run build

# install types
pnpm add -D @types/testing-library__jest-dom

# runs well
pnpm
```

Question:
why should i install `@types/testing-library__jest-dom`? it just is a dependency of `@testing-library/jest-dom`

infact, `@types/testing-library__jest-dom` has been installed at .pnpm but can't be found by typescript:

![installed](https://user-images.githubusercontent.com/16874003/167984587-2b183a40-a086-4a16-91bc-22d6f3e58ea0.png)

![not in @types unless you install it](https://user-images.githubusercontent.com/16874003/167984594-8c0bd5b8-28cf-4a39-95f6-dfb2ed9be3e1.jpg)

**Comment by tiagobento** [comment]
I might be wrong, but I think `@types/testing-library__jest-dom` is a transitive dependency coming from `@testing-library/jest-dom`, right? This means that only code inside `@testing-library/jest-dom` will have direct access to `@types/testing-library__jest-dom`. 

This is done so to prevent you from depending directly on a transitive dependency, and to help `pnpm` correctly resolve peerDependencies too.

If your code is importing `@testing-library/jest-dom`, then you should list `@types/testing-library__jest-dom` on your package.json. This will create a symlink on your node_modules directory pointing to the dependency, which will work with the standard node module resolution from TypeScript.

**Reply by juilletVent** [reply]
great,is worked
